docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -911,6 +911,37 @@ The methods are available because the `column` argument is parsed as a SQLGlot [
 
 Column expressions are sub-classes of the [Condition class](https://sqlglot.com/sqlglot/expressions.html#Condition), so they have builder methods like [`between`](https://sqlglot.com/sqlglot/expressions.html#Condition.between) and [`like`](https://sqlglot.com/sqlglot/expressions.html#Condition.like).
 
+#### Accessing model schemas
+
+Model schemas can be accessed within a Python macro function through its evaluation context's `column_to_types` method, when they can be statically determined.
+
+For example, consider the following macro function which aims to rename the columns of a target model by adding a prefix to them:
+
+```python linenums="1"
+from sqlglot import exp
+from sqlmesh.core.macros import macro
+
+@macro()
+def foo(evaluator, model_name, prefix):
+    prefix = prefix.name
+    renamed_projections = []
+
+    for name, dtype in evaluator.columns_to_types(model_name.sql()).items():
+        new_name = prefix + name
+        renamed_projections.append(exp.cast(exp.column(name), dtype).as_(new_name))
+
+    return renamed_projections
+```
+
+Having access to the schema of an upstream model can be useful for various reasons:
+
+- Normalizing columns so that downstream consumers are not tightly coupled e.g. to external or source tables
```

Comment:
What does "normalizing" mean in this context?

Reply:
Anything really. One way to normalize them is to add prefixes as shown in the exampe, another is to transform `__` to `_` as done in an example Alex had shared. The point is to transform them in some way that establishes a naming protocol for downstream consumers.

Reply:
Maybe use "Renaming" to avoid ambiguity with normalizing/denormalizing tables?

Reply:
> Maybe use "Renaming" to avoid ambiguity with normalizing/denormalizing tables?

Sounds good

Reply:
Addressed